src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.threshold = 12;
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
+
+        int indexOfBucket = hash(key);
+
+        if (table[indexOfBucket] == null) {
+            table[indexOfBucket] = new Node(key, value);
+            size++;
+        }
 
+        Node<K, V> pointer = table[indexOfBucket];
+
+        while (pointer != null) {
+            if (pointer.key == key || key != null && key.equals(pointer.key)) {
+                pointer.value = value;
+                return;
+            }
+            if (pointer.next == null) {
+                pointer.next = new Node(key, value);
+                size++;
+                return;
+            }
+            pointer = pointer.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexOfBucket = hash(key);
+        Node<K, V> pointer = table[indexOfBucket];
+
+        while (pointer != null) {
+            if (pointer.key == key || key != null && key.equals(pointer.key)) {
```

Comment:
you have the same condition a second time. extract it to function

Reply:
not fixed